flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/Codedata.java

```diff
@@ -35,11 +36,12 @@
  * @param parentSymbol The parent symbol of the component
  * @param resourcePath The path of the resource function
  * @param id           The unique identifier of the component if exists
+ * @param kind         Type kind of the component
  * @since 2.0.0
  */
 public record Codedata(NodeKind node, String org, String module, String object, String symbol,
                        String version, LineRange lineRange, String sourceCode, String parentSymbol,
-                       String resourcePath, Integer id) {
+                       String resourcePath, Integer id, String kind) {
```

Comment:
+1

Reply:
As a second thought, I feel like having a NodeKind is not suitable for a type since its a type-definition and its outside of the flow-model. @hasithaa  any thoughts?

Reply:
We should rename NodeKind to a more generic term, since this name originated when the flow model was the only model using codedata. Adding another field for the same purposes would unnecessarily bloat the JSON.